### ISSUE TITLE
LaTeX Error: File ... not found.

### ISSUE THREAD

**Issue by simevo** [doxygen/doxygen#8296]
**Describe the bug**
while [rebuilding most of the debian packages that use doxygen during the build](https://salsa.debian.org/debian/doxygen/-/wikis/ratt_doxygen_1.9.0-1_amd64-partial), 5 packages failed to build with doxygen 1.9.0 (that did not happen with 1.8.20) because of a weird LaTeX Error:
- caffe: `! LaTeX Error: File 'classBlockingQueue_3_01caffe_1_1Batch_3_01Dtype_01_4_01_5_01_4_1_1sync.tex' not found.`
- gdcm: `! LaTeX Error: File 'classbasic__string_3_01char_01_4_1_1const__iterator.tex' not found.`
- gr-radar: `! LaTeX Error: File 'classbasic__string_3_01char_01_4_1_1const__iterator.tex' not found.`
- sdformat: `! LaTeX Error: File 'classbasic__string_3_01char_01_4_1_1const__iterator.tex' not found.`
- uhd:`! LaTeX Error: File 'classbasic__string_3_01char_01_4_1_1const__iterator.tex' not found.`

**Expected behavior**
works fine with 1.8.20

**Screenshots**

**To Reproduce**
for example with `gr-radar`:
```
dget https://deb.debian.org/debian/pool/main/g/gr-radar/gr-radar_0.0.0.20180308-4.dsc
cd gr-radar-0.0.0.20180308#
apt install -t experimental doxygen doxygen-latex
apt build-dep gr-radar
dpkg-buildpackage -uc -us
```

I attach the contents of the `gr-radar-0.0.0.20180308/obj-x86_64-linux-gnu/docs/doxygen` dir at the end of the build: 
[doxygen.zip](https://github.com/doxygen/doxygen/files/5759471/doxygen.zip)

**Version**
1.9.0 from Debian experimental

**Stack trace**

**Additional context**

**Comment by albert-github** [comment]
@doxygen I ran with the current master (1.9.1 (75e96744145e9e21f28691e6d0c9b16523050356)) and got a corrupted stack:
```
Generating code for file mmap.h...
Generating code for file mmap.test.h...
Parsing code for file mmap_v2.cpp...

Thread 1 "doxygen" received signal SIGSEGV, Segmentation fault.
0x0000000000000000 in ?? ()
(gdb) where
#0  0x0000000000000000 in ?? ()
Backtrace stopped: previous frame identical to this frame (corrupt stack?)
```

**Comment by albert-github** [comment]
I can see that we have here a problem.

Problem is I cannot run the examples as I don't have a Debian system and thus cannot run the mentioned commands.
- It would be useful to have a pointer to the (original) source packages.

Looking in the `latex/refman.tex` I see a number of `classbasic...`, `classvector...` and `classgr...` input statements, though on the directory there are only `classgr...tex` files.
Looking at the HTML page `html/classes.html` I see here an entry `basic_string< char16_t >::const_iterator` but when clicking on it it gives a "Problem loading" for file "...html/classbasic__string_3_01char16__t_01_4_1_1const__iterator.html.".

(@doxygen might this have to do with `std::` classes?)

**Comment by simevo** [comment]
here are the upstream pointers:
- http://caffe.berkeleyvision.org/
- http://gdcm.sourceforge.net/
- https://grradar.wordpress.com/, https://github.com/kit-cel/gr-radar
- http://sdformat.org/, https://github.com/osrf/sdformat
- https://www.ettus.com/sdr-software/uhd-usrp-hardware-driver/, https://github.com/EttusResearch/uhd

**Comment by albert-github** [comment]
Looks like I can reproduce the problem as well with the doxygen's own internal documentation (Doxyfile on the main directory and output in doxygen_docs) by setting the `GENERATE=LATEX=YES` and `BUILTIN_STL_SUPPORT    = YES`, in this case I see also some references to `classvector...`  in the refman.tex but no files `calssvector...`.

**Comment by doxygen** [comment]
@simevo Please verify if the referenced commit fixes the problem for you.

**Comment by albert-github** [comment]
I tested it with the internal doxygen documentation and this looks OK now.

**Comment by simevo** [comment]
Hi ! it seems to fix it for all above listed packages except caffe. It could be an issue with that one.

I reproduce with:
```
git clone https://github.com/BVLC/caffe
cd caffe
doxygen .Doxyfile 
cd doxygen/latex/
pdflatex refman
```

errors:
```
! LaTeX Error: File `classBlockingQueue_3_01caffe_1_1Batch_3_01Dtype_01_4_01_5_01_4_1_1sync.tex' not found.

Type X to quit or <RETURN> to proceed, or enter new name. (Default extension: tex)
```

I attach the contents of the `doxygen/latex` dir at the end of the run:
[latex.zip](https://github.com/doxygen/doxygen/files/5764443/latex.zip)

**Comment by albert-github** [comment]
@simevo @doxygen As far as I know the issues in this issue are all solved except for the caffe issue, i.e. the reference to `classBlockingQueue_3_01caffe_1_1Batch_3_01Dtype_01_4_01_5_01_4_1_1sync.tex`

**Comment by albert-github** [comment]
I tested the caffe project and the LaTeX build builds (so does the rtf version, was just an extra test).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.1.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
